app/src/main/java/org/mifos/mobile/api/DataManager.kt

```diff
@@ -0,0 +1,247 @@
+package org.mifos.mobile.api
+
+import io.reactivex.Observable
+import io.reactivex.ObservableSource
+import io.reactivex.functions.Function
+import okhttp3.MediaType
+import okhttp3.ResponseBody
+import org.mifos.mobile.FakeRemoteDataSource
+import org.mifos.mobile.api.local.DatabaseHelper
+import org.mifos.mobile.api.local.PreferencesHelper
+import org.mifos.mobile.models.*
+import org.mifos.mobile.models.accounts.loan.LoanAccount
+import org.mifos.mobile.models.accounts.loan.LoanWithAssociations
+import org.mifos.mobile.models.accounts.loan.LoanWithdraw
+import org.mifos.mobile.models.accounts.savings.SavingsAccountApplicationPayload
+import org.mifos.mobile.models.accounts.savings.SavingsAccountUpdatePayload
+import org.mifos.mobile.models.accounts.savings.SavingsAccountWithdrawPayload
+import org.mifos.mobile.models.accounts.savings.SavingsWithAssociations
+import org.mifos.mobile.models.beneficiary.Beneficiary
+import org.mifos.mobile.models.beneficiary.BeneficiaryPayload
+import org.mifos.mobile.models.beneficiary.BeneficiaryUpdatePayload
+import org.mifos.mobile.models.client.Client
+import org.mifos.mobile.models.client.ClientAccounts
+import org.mifos.mobile.models.guarantor.GuarantorApplicationPayload
+import org.mifos.mobile.models.guarantor.GuarantorPayload
+import org.mifos.mobile.models.guarantor.GuarantorTemplatePayload
+import org.mifos.mobile.models.notification.MifosNotification
+import org.mifos.mobile.models.notification.NotificationRegisterPayload
+import org.mifos.mobile.models.notification.NotificationUserDetail
+import org.mifos.mobile.models.payload.LoansPayload
+import org.mifos.mobile.models.payload.LoginPayload
+import org.mifos.mobile.models.payload.TransferPayload
+import org.mifos.mobile.models.register.RegisterPayload
+import org.mifos.mobile.models.register.UserVerify
+import org.mifos.mobile.models.templates.account.AccountOptionsTemplate
+import org.mifos.mobile.models.templates.beneficiary.BeneficiaryTemplate
+import org.mifos.mobile.models.templates.loans.LoanTemplate
+import org.mifos.mobile.models.templates.savings.SavingsAccountTemplate
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/**
+ * @author Vishwajeet
+ * @since 13/6/16.
+ */
+@Singleton
+class DataManager @Inject constructor(val preferencesHelper: PreferencesHelper, private val baseApiManager: BaseApiManager,
```

Comment:
Since, you have converted this class to Kotlin, just check whether API call, throughout the app, are working fine, by manually testing the app since we don't have UI tests as of now.

Reply:
I tested it manually. Every API calls which worked until the previous commit in the development branch did work fine.

Reply:
it is throwing error in importing FakeRemoteDataSource // line 8

Reply:
@Him887 That error is thrown by IDEA inspection. The tests are running absolutely fine and without any issues. There is no universal fix for this unless JetBrains fixes it in a future update. [This](https://discuss.kotlinlang.org/t/accessing-internal-class-from-java-tests-in-the-same-module/7611/2) might throw more light to you on this issue

[https://discuss.kotlinlang.org/t/accessing-internal-class-from-java-tests-in-the-same-module/7611/2](https://discuss.kotlinlang.org/t/accessing-internal-class-from-java-tests-in-the-same-module/7611/2)